spacelift/resource_drift_detection.go

```diff
@@ -0,0 +1,171 @@
+		return diag.Errorf("could not create drift detection integration for the stack: %v", err)
+	}
+
+	if d.Id() == "" {
+		d.SetId(stackID)
+	}
+
+	return resourceDriftDetectionRead(ctx, d, meta)
+}
+
+func resourceDriftDetectionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		CreateDriftDetectionIntegration struct {
+			Reconcile bool     `graphql:"reconcile"`
+			Schedule  []string `graphql:"schedule"`
+		} `graphql:"stackIntegrationDriftDetectionUpdate(stack: $stack, input: $input)"`
+	}
+
+	var scheduleExpressions []graphql.String
+	for _, expr := range d.Get("schedule").([]interface{}) {
+		scheduleExpressions = append(scheduleExpressions, graphql.String(expr.(string)))
+	}
+
+	stackID := d.Get("stack_id").(string)
+
+	variables := map[string]interface{}{
+		"stack": toID(stackID),
+		"input": structs.DriftDetectionIntegrationInput{
+			Reconcile: graphql.Boolean(d.Get("reconcile").(bool)),
+			Schedule:  scheduleExpressions,
+		},
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "DriftDetectionCreate", &mutation, variables); err != nil {
+		return diag.Errorf("could not update drift detection integration for the stack: %v", err)
+	}
+
+	if d.Id() == "" {
+		d.SetId(stackID)
+	}
+
+	return resourceDriftDetectionRead(ctx, d, meta)
+}
+
+func resourceDriftDetectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceStackDriftDetectionReadWithHooks(ctx, d, meta, func(_ string) diag.Diagnostics {
+		d.SetId("")
+		return nil
+	})
+}
+
+func resourceDriftDetectionDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		DeleteDriftDetectionIntegration struct {
+			Deleted bool `graphql:"deleted"`
+		} `graphql:"stackIntegrationDriftDetectionDelete(stack: $stack)"`
+	}
+
+	variables := map[string]interface{}{"stack": toID(d.Id())}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "DriftDetectionDelete", &mutation, variables); err != nil {
+		return diag.Errorf("could not delete drift detection integration for stack: %v", err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func resourceStackDriftDetectionReadWithHooks(ctx context.Context, d *schema.ResourceData, meta interface{}, onNil func(message string) diag.Diagnostics) diag.Diagnostics {
+	var query struct {
+		Stack *structs.Stack `graphql:"stack(id: $id)"`
+	}
+
+	variables := map[string]interface{}{"id": toID(d.Id())}
+
+	if err := meta.(*internal.Client).Query(ctx, "StackDriftDetectionRead", &query, variables); err != nil {
+		return diag.Errorf("could not query for stack: %v", err)
+	}
+
+	if query.Stack == nil {
+		return onNil("stack not found")
+	}
+
+	integration := query.Stack.Integrations.DriftDetection
+
+	d.Set("reconcile", integration.Reconcile)
+
+	schedule := make([]interface{}, len(integration.Schedule))
+	for i, expr := range integration.Schedule {
+		schedule[i] = expr
+	}
+	d.Set("schedule", schedule)
```

Comment:
I think Terraform SDK docs suggest that you should check for the error when calling `Set` with a collection type (list, set, map) argument.